ViSearchSDK/ViSearchSDK/Classes/Response/ProductSearch/ViProductSearchResponse.swift

```diff
@@ -0,0 +1,206 @@
+            result.message = message
+        }
+        
+        return result
+    }
+     
+    /// Returns an array of ViProduct that is parsed from an array of json object
+    ///
+    /// - parameter arr: An array of json object for the "result" field
+    ///
+    /// - returns: An array of ViProducts found by the server
+    private static func parseProductResults(_ arr: [Any]) -> [ViProduct] {
+        var results : [ViProduct] = []
+        // iterate through each json object
+        for jsonItem in arr {
+            // convert from json object to dictionary and then parse the data
+            if let dict = jsonItem as? [String:Any] {
+                
+                let item = ViProduct()
+                
+                if let productId = dict["product_id"] as? String {
+                    item.productId = productId
+                }
+                
+                if let mainImageUrl = dict["main_image_url"] as? String {
+                    item.mainImageUrl = mainImageUrl
+                }
+                
+                if let data = dict["data"] as? [String:Any] {
+                    item.data = data
+                }
+                
+                if let score = dict["score"] as? Float {
+                    item.score = score
+                }
+                
+                if let imageS3Url = dict["image_s3_url"] as? String {
+                    item.imageS3Url = imageS3Url
+                }
+                
+                if let detect = dict["detect"] as? String {
+                    item.detect = detect
+                }
+                
+                if let keyword = dict["keyword"] as? String {
+                    item.keyword = keyword
+                }
+                
+                if let box = dict["box"] as? [Int] {
+                    item.box = ViBox(x1: box[0], y1: box[1], x2: box[2], y2: box[3])
+                }
+                
+                results.append(item)
+            }
+        }
+        return results
+    }
+    
+    /// Returns an array of ViGroupResult that is parsed from an array of json object
+    ///
+    /// - parameter arr: An array of json object for the "group_results" field
+    ///
+    /// - returns: An array of ViGroupResult grouped by the server
+    private static func parseGroupResults(_ arr: [Any]) -> [ViGroupResult] {
+        var results: [ViGroupResult] = []
+        // iterate through each json object
+        for jsonItem in arr {
+            // convert from json object to dictionary and then parse the data
+            if let dict = jsonItem as? [String:Any] {
+                
+                let group = dict["group_by_value"] as! String
+                
+                let item = ViGroupResult(group: group)!
+                
+                if let imageResults = dict["results"] as? [Any] {
+                    item.results = ViResponseData.parseResults(imageResults)
```

Comment:
cannot reuse, API response is different. ViSearch is having value_map, Product Search is `data`